### PR TITLE
Salesforce book names

### PR DESCRIPTION
**Pull request by Dantemss**
Closes https://github.com/openstax/tutor-server/issues/861

- Populate book names from SF
- Results are cached
- Cache is regenerated every day, when we update the Salesforce stats
- (Some) specs

### CODE REVIEW

**Comment by Dantemss** [comment]
As a side note, we won't be able to create/edit course offerings in dev, but maybe we don't care...

File: config/secrets.yml.example [hunk 1]
```diff
@@ -32,6 +32,7 @@ development:
       cache: <%= ENV["REDIS_NAMESPACES_CACHE"] || 'cache' %>
       settings: <%= ENV["REDIS_NAMESPACES_SETTINGS"] || 'settings' %>
       jobba: <%= ENV["REDIS_NAMESPACES_LEV"] || 'jobba' %>
+      active_force: <%= ENV["REDIS_NAMESPACES_ACTIVE_FORCE"] || 'active_force' %>
```
Comment (jpslav) [inline config/secrets.yml.example]: Need a corresponding tutor-deployment PR?

File: app/views/admin/catalog_offerings/_edit_row.html.erb [hunk 2]
```diff
@@ -9,11 +9,12 @@
 <div class='offering-row'>
   <div class="form-group">
     <%= form.label :salesforce_book_name %>
-    <%= form.text_field :salesforce_book_name,  {class: 'form-control'} %>
+    <%= form.select :salesforce_book_name, @salesforce_book_names,
+                    { include_blank: true }, {class: 'form-control'} %>
```
Comment (jpslav) [inline app/views/admin/catalog_offerings/_edit_row.html.erb]: To enable course offering creation and editing in dev/QA, how about:

```ruby
<% if is_real_production_site? %>
  <%= form.text_field :salesforce_book_name,  {class: 'form-control'} %>
<% else %>
  <%= form.select :salesforce_book_name, @salesforce_book_names,
                  { include_blank: true }, {class: 'form-control'} %>
<% end %>
```

File: config/secrets.yml.example (hunk 1 above)
Comment (Dantemss) [inline config/secrets.yml.example]: I thought there was but apparently not... yeah we need one.

File: app/views/admin/catalog_offerings/_edit_row.html.erb (hunk 2 above)
Comment (Dantemss) [inline app/views/admin/catalog_offerings/_edit_row.html.erb]: Sure, I can do that.

**Pull request merged**